File: app/src/main/java/com/sumian/hw/upgrade/dialog/Version2ConnectingDialog.java
package com.sumian.hw.upgrade.dialog;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import com.sumian.hw.widget.BaseDialogFragment;
import com.sumian.sd.R;

import butterknife.ButterKnife;

/**
 * DFU 升级时,监测仪断开,速眠仪连接阶段的 dialog
 */
public class Version2ConnectingDialog extends BaseDialogFragment {

    public static Version2ConnectingDialog newInstance() {
        return new Version2ConnectingDialog();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NO_TITLE, R.style.SumianDialog);
    }

    @SuppressWarnings("ConstantConditions")
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        getDialog().getWindow().setBackgroundDrawableResource(android.R.color.transparent);
        getDialog().requestWindowFeature(Window.FEATURE_NO_TITLE);
        View rootView = inflater.inflate(getLayout(), container, false);
        ButterKnife.bind(this, rootView);
        initView(rootView);
        return rootView;
    }

    @Override
    protected int getLayout() {
        setCancelable(false);
        return R.layout.hw_lay_dialog_version_connectting;
    }
}

File: app/src/main/java/com/sumian/hw/upgrade/dialog/Version2ConnectingDialog.kt
package com.sumian.hw.upgrade.dialog

import android.os.Bundle
import androidx.fragment.app.DialogFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window

import com.sumian.hw.widget.BaseDialogFragment
import com.sumian.sd.R


/**
 * DFU 升级时,监测仪断开,速眠仪连接阶段的 dialog
 */
class Version2ConnectingDialog : BaseDialogFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(androidx.fragment.app.DialogFragment.STYLE_NO_TITLE, R.style.SumianDialog)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        dialog.window!!.setBackgroundDrawableResource(android.R.color.transparent)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        val rootView = inflater.inflate(layout, container, false)
        initView(rootView)
        return rootView
    }

    override fun getLayout(): Int {
        isCancelable = false
        return R.layout.hw_lay_dialog_version_connectting
    }

    companion object {

        fun newInstance(): Version2ConnectingDialog {
            return Version2ConnectingDialog()
        }
    }
}
